extensions/spring-data-jpa/deployment/src/main/java/io/quarkus/spring/data/deployment/generate/SpringDataRepositoryCreator.java

```diff
@@ -142,8 +143,15 @@ public void implementCrudRepository(ClassInfo repositoryToImplement) {
         }
 
         if (idDotName == null || entityDotName == null) {
-            throw new IllegalArgumentException(
-                    "Repository " + repositoryToImplement + " does not specify ID and/or Entity type");
+            AnnotationInstance repositoryDefinitionInstance = repositoryToImplement
+                    .classAnnotation(DotNames.SPRING_DATA_REPOSITORY_DEFINITION);
+            if (repositoryDefinitionInstance != null) {
+                entityDotName = repositoryDefinitionInstance.value("domainClass").asClass().name();
+                idDotName = repositoryDefinitionInstance.value("idClass").asClass().name();
```

Comment:
I'm not sure about the place of this. If `@RepositoryDefinition` is used, can't we skip all the code above?

Reply:
You are absolutely right! Fixed